new-frontiers-14/frontier-station-14
# Add a component that can make doors airtight, for player made docking ports & EVA access

## About the PR


Added a new component & corresponding system, AdvDoorSeals. This component:
- Checks if the door has power
- Checks if the door is facing space (like a thruster)
- If these two checks pass, makes the door airtight like a directional fan
- If either of these conditions stops being true, lets air pass. Don't lose power while docked!

## Why / Balance

So that constructs and other ship extensions no longer have to choose between being pressurised and having only 1 wide docks.
Checks are in place to ensure it's not abusable by atmos. It may be abusable by ship builders as it is effectively a buildable directional fan that can be placed only at the edge of a grid and needs power, but that can be balanced by upping power draw on any entities this gets added to.

## How to test

Add the component to a door (currently not added to anything in yml)
Check that it blocks airflow when powered and the tile it's facing is space (either empty or lattice)
Check that it allows airflow when not powered or a solid tile is in it's facing direction (to prevent atmos shenanigans)

## Media



https://github.com/user-attachments/assets/b7f9b92b-e1f7-479d-80d1-a940ecbc3dc2


## Requirements



## Breaking changes

Some code was pulled out of thrusters and put into shuttles, this is to make the tileupdating work by making the spaced/not spaced check an event. This also inadvertently fixes the bug where pulling up plating doesn't update thruster nozzle space exposure checks. Not sure if that counts as a breaking change, but it does touch upstream code a decent amount.

**Changelog**

:cl:
- add: Added airtight seals that can be added to doors through admeme magic
